docs/USAGE_PATTERNS.md

```diff
@@ -0,0 +1,87 @@
+# Usage Patterns
+
+`fulcrum-core` offers two APIs that coexist: the original API for backward compatibility and a new, modern API for TypeScript-first development. You can use them separately or together.
+
+---
+
+## 1. Original API (Default)
+
+This is the classic API that remains fully functional. Use it to maintain existing code with minimal changes.
+
+**Use Case**: Maintaining existing applications.
+
+```javascript
+import { Form, Record } from '@fulcrumapp/fulcrum-core';
+
+// Create and load a form using the ActiveRecord-style pattern
+const form = new Form(attributes);
+await form.load(dataSource);
+
+// All original methods work as before
+console.log(form.name);
+```
+
+-   **Pros**: No migration effort, familiar patterns.
+-   **Cons**: Lacks clear separation of concerns, less testable.
+
+---
+
+## 2. Modern API
+
+The modern API provides pure, TypeScript-first models and services, promoting a clean architecture.
+
+**Use Case**: New projects or refactoring existing features.
+
+```typescript
+import { FormModel, FormService } from 'fulcrum-core/modern';
+import type { FormAttributes } from 'fulcrum-core/types';
+
+// Create a pure data model
+const form = new FormModel(attributes);
+
+// Use a service to perform actions
+await FormService.load(form, dataSource);
+
+console.log(form.name); // Full TypeScript support
+```
+
+-   **Pros**: Strong typing, separation of concerns, improved testability.
+-   **Cons**: Requires adopting a new pattern.
+
+---
+
+## 3. Types Only
```

Comment:
We don't need to and shouldn't recommend "types only" usage - the idea is to provide a uniform representation of our data models. That means that everywhere should use the same thing